staging/src/k8s.io/client-go/tools/pager/pager.go

```diff
@@ -115,3 +127,86 @@ func (p *ListPager) List(ctx context.Context, options metav1.ListOptions) (runti
 		options.Continue = m.GetContinue()
 	}
 }
+
+// EachListItem invokes fn on each runtime.Object in the list. Any error immediately terminates the
+// loop. The items are retrieved via List in small chunks from the server to reduce the impact on
+// the server. Up to ListPager.PageBufferSize chunks are buffered concurrently in the background.
+// If the chunk attempt fails a "Expired" error may be returned.
+func (p *ListPager) EachListItem(ctx context.Context, options metav1.ListOptions, fn func(obj runtime.Object) error) error {
+	return p.eachListChunkBuffered(ctx, options, func(obj runtime.Object) error {
+		return meta.EachListItem(obj, fn)
+	})
+}
+
+// stoppedErr is an internal error for signaling eachListChunk should stop processing chunks.
+// It must be handled internally and never be returned to API callers.
+var stoppedErr = goerrors.New("paginated list stopped")
+
+// eachListChunkBuffered invokes fn on each runtimeObject list chunk. It buffers up to
+// PageBufferSize chunks of list items concurrently in the background. If the chunk attempt fails a "Expired"
+// error may be returned.
+func (p *ListPager) eachListChunkBuffered(ctx context.Context, options metav1.ListOptions, fn func(obj runtime.Object) error) error {
+	chunkC := make(chan runtime.Object, p.PageBufferSize)
+
+	stopC := make(chan struct{})
+	bgResultC := make(chan error, 1)
+	go func() {
```

Comment:
`defer runtime.HandleCrash()` at the top of the goroutine?

Reply:
Good catch, I should do something with panics.  I've gone with `runtime.RecoverFromPanic` here. Hopefully it's appropriate.

Reply:
`runtime.RecoverFromPanic` didn't work as I had expected (and from a quick grep, doesn't look like we use it in the k8s codebase), so I did a more direct recover.

Reply:
Why you can't use runtime.HandleCrash() - it's pretty widely used in the codebase and allows to use additional custom handler too.

Reply:
I had initially misunderstood it to catch the panic and swallow it. But it does still crash. Updating the PR to use it now.